### ISSUE TITLE
Display tax percent in current locale number format

### ISSUE THREAD

**Issue by daniel-ifrim** [magento/magento2#18724]
### Preconditions (*)

1. Magento 2.4-develop
2. Set locale to Germany or other European country where decimal character is comma
3. In Admin > Stores > Configuration > Sales > Tax > Shopping Cart Display Settings > Display Full Tax Summary = Yes and in the other section "Orders, Invoices, Credit Memos Display Settings"
4. Set a tax percent in admin with decimals like 50.1.

### Steps to reproduce (*)

1. Add in cart a product.
2. Go to cart page
3. Go to checkout and look at the summary block
4. Place the order and see it in My Account
5. See the order in admin
6. See emails sent about order

### Expected result (*)

1. It should display "50,1" percent with comma.

### Actual result (*)

1. In cart estimation and in other places displays "50.1" percent
2. The same number format in other places
In email
![image](https://user-images.githubusercontent.com/51680850/108998908-3fb0c780-76aa-11eb-896c-bf1b21d46e8b.png)
In summary block
![image](https://user-images.githubusercontent.com/51680850/109010438-7097f900-76b8-11eb-87d6-b8a1e7e9c610.png)

**Comment by magento-engcom-team** [comment]
Hi @daniel-ifrim. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@daniel-ifrim do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by daniel-ifrim** [comment]
@magento-engcom-team give me 2.3

**Comment by daniel-ifrim** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @daniel-ifrim. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @daniel-ifrim, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18724-2-3-develop/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18724-2-3-develop//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18724-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by daniel-ifrim** [comment]
@magento-engcom-team I managed to add to cart on the instance given by adding the product to comparison list and than add to cart. Otherwise it says Session has expired or something.

**Comment by Shubham0492** [comment]
#SQUASHTOBERFEST

**Comment by magento-engcom-team** [comment]
Hi @Shubham-Webkul. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by Shubham0492** [comment]
magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
@shikhamis11 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97301` were created

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-41036` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by danielrussob** [comment]
@magento give me test instance

**Comment by danielrussob** [comment]
@magento give me 2.4-develop instance

**Comment by danielrussob** [comment]
Hello @daniel-ifrim ,
thank you for your contribution

I'm not able to see a fix on your PR's instance, as you can see in the screenshot

![image](https://user-images.githubusercontent.com/13854502/117855736-0fc99700-b28b-11eb-9174-88caa9e52490.png)

**Comment by engcom-Delta** [comment]
Hi @daniel-ifrim ,

Verified the issue on 2.4-develop and it is reproducible.

Hence confirming this issue

Preconditions (*)


1.Set locale to Germany or other European country where decimal character is comma
2.In Admin > Stores > Configuration > Sales > Tax > Shopping Cart Display Settings > Display Full Tax Summary = Yes and in 
   the other section "Orders, Invoices, Credit Memos Display Settings"
3.Set a tax percent in admin with decimals like 50.1.

Steps to reproduce:-

1.Add in cart a product.
2.Go to cart page
3.Go to checkout and look at the summary block
4.Place the order and see it in My Account
5.See the order in admin
6.See emails sent about order

Set locale to Germany

<img width="1431" alt="Screenshot 2024-05-29 at 6 27 40 PM" src="https://github.com/magento/magento2/assets/51681379/5f592eb2-d494-452a-9f93-ca0ff0b5a841">

In Admin > Stores > Configuration > Sales > Tax > Shopping Cart Display Settings > Display Full Tax Summary = Yes 

<img width="1435" alt="Screenshot 2024-05-29 at 6 28 28 PM" src="https://github.com/magento/magento2/assets/51681379/6363fc1d-905b-40dc-80bd-fe1b525359eb">

"Orders, Invoices, Credit Memos Display Settings"

<img width="1438" alt="Screenshot 2024-05-29 at 6 29 18 PM" src="https://github.com/magento/magento2/assets/51681379/83241a20-bfc9-49e8-a0c9-da4fb5455af6">

Add in cart a product

<img width="1440" alt="Screenshot 2024-05-29 at 6 33 27 PM" src="https://github.com/magento/magento2/assets/51681379/617fcf4c-4111-44f3-8a88-0b40579984fa">

Go to checkout and look at the summary block

<img width="1436" alt="Screenshot 2024-05-29 at 6 33 35 PM" src="https://github.com/magento/magento2/assets/51681379/8564d631-aef6-47d3-933f-1a0e8750af54">

<img width="18" alt="Screenshot 2024-05-29 at 6 33 50 PM" src="https://github.com/magento/magento2/assets/51681379/427ca21e-8457-4476-b5d1-f8a5c55379f0">

<img width="1436" alt="Screenshot 2024-05-29 at 6 34 06 PM" src="https://github.com/magento/magento2/assets/51681379/6a6c5833-6f42-4dd5-8f62-23430b988d06">

<img width="1431" alt="Screenshot 2024-05-29 at 6 38 39 PM" src="https://github.com/magento/magento2/assets/51681379/a7f69b55-463a-49d8-a29e-6e9b8137b324">

<img width="1431" alt="Screenshot 2024-05-29 at 6 46 47 PM" src="https://github.com/magento/magento2/assets/51681379/8d136d5f-7c92-4b16-8c70-e21f333c879f">

<img width="1436" alt="Screenshot 2024-05-29 at 6 47 22 PM" src="https://github.com/magento/magento2/assets/51681379/46e13d4a-c5a5-4edc-b1fa-5456cf467604">

In cart estimation and in other places displays "50.1" percent

<img width="1429" alt="Screenshot 2024-05-29 at 6 49 59 PM" src="https://github.com/magento/magento2/assets/51681379/611d1ac2-2fa8-44c9-8537-f766387947d2">

but in admin product details it is not as german locale

<img width="1435" alt="Screenshot 2024-05-29 at 6 51 10 PM" src="https://github.com/magento/magento2/assets/51681379/063c93a2-7765-4540-ac6f-ee920f710786">

**Comment by Sathakathulla-S** [comment]
@magento I'm working on this
